doc/user/content/sql/tail.md

```diff
@@ -328,33 +309,115 @@ while (true)
-  await client.connect();
+  conn.setAutoCommit(false);
+
+  stmt.executeUpdate("DECLARE c CURSOR FOR TAIL mz_scheduling_elapsed;");
 
-  await client.query('BEGIN');
-  await client.query('DECLARE c CURSOR FOR TAIL t');
   while (true) {
-    const res = await client.query('FETCH ALL c');
-    console.log(res.rows);
+      ResultSet fetchResultSet = stmt.executeQuery("FETCH 100 c WITH (timeout='1s');");
+
+      while (fetchResultSet.next()) {
+          System.out.println(fetchResultSet.getString(3));
+      }
   }
-}
+```
+
+| Additional guides |
+| ---------------------- |
+| [Node.js](/guides/node-js/#stream)|
+| [PHP](/guides/php-pdo/#stream)|
+| [GO](/guides/golang/#stream) |
+### Using `AS OF`
 
-main();
+Start Materialize with a custom compaction window [`--logical-compaction-window 10000`](/cli/#compaction-window) and create a non-materialized view:
+
+```sql
+CREATE VIEW most_scheduled_worker AS
+  SELECT
+    worker,
+    SUM(elapsed_ns) as time_working
+  FROM mz_scheduling_elapsed
+  GROUP BY worker
+  ORDER BY time_working DESC
+  LIMIT 1;
 ```
 
-### Interactive `TAIL`
+Create an index and set a the compaction:
 
-If you want to use `TAIL` from an interactive SQL session (for example in `psql`), wrap the query in `COPY`.
+```sql
+CREATE INDEX most_scheduled_worker_idx
+  ON most_scheduled_worker (worker, time_working)
+  WITH (logical_compaction_window = '10 seconds');
+```
+
+Stream out changes from ten seconds before the statement is executed:
 
 ```sql
-COPY (TAIL t) TO STDOUT
+COPY (TAIL most_scheduled_worker AS OF NOW() - INTERVAL '10 seconds') TO STDOUT;
 ```
 
-[`bigint`]: /sql/types/bigint
-[`numeric`]: /sql/types/numeric
+Take into account, for this example, that ten logical seconds need to pass by inside Materialize to browse and recover changes from the last ten seconds.
+
+### Relating rows with their updates
+
+After all the rows from the `SNAPSHOT` have been transmitted, the updates will come as they occur,
+but how is it possible to know which row has been updated? <br/>
+
+| mz_timestamp | mz_progressed | mz_diff | Column 1 | .... | Column N |
+| ------------ | ------------- | ------- | -------- | ---- | -------- | --------------------------- |
+| 1            | false         | 1       | id1      |      | value1   |
+| 1            | false         | 1       | id2      |      | value2   |
+| 1            | false         | 1       | id3      |      | value3   | <- Last row from `SNAPSHOT` |
+| 2            | false         | -1      | id1      |      | value1   |
+| 2            | false         | 1       | id1      |      | value4   |
+
+If you already know your data, then any _column_ acting as a key will do the work but if the key is unknown, then using a `hash(columns_values)` could serve as your key. <br/><br/>
+Back to the example, `Column 1` acts as the key to know from where the update comes from; in case this was unknown, hashing the values from `Column 1` to `Column N` should reveal the origin row.
```

Comment:
```suggestion
If your data has a unique key, this can be used to map an update to its corresponding row; if the key is unknown, you can use the output of `hash(columns_values)` instead. <br/><br/>
In the example above, `Column 1` acts as the key that uniquely identifies the row the update refers to; in case this was unknown, hashing the values from `Column 1` to `Column N` would identify the origin row.
```